R/ospPlotConfiguration.R

```diff
@@ -0,0 +1,403 @@
+#' @title Convenience class for plot configuration for OSP plots
+#'
+#' @description
+#'
+#' R6 class defining the configuration for `{ospsuite}` plots. It holds values
+#' for all relevant plot properties.
+#'
+#' Note that the values this objects contains are of general-purpose utility. In
+#' other words, the public members of this class instance can be used to specify
+#' arguments for base plots, ggplot plots, or any other plotting framework.
+#'
+#' @field title,subtitle,xlabel,ylabel,legendTitle,watermark A character string
+#'   providing plot annotations for plot title, subtitle, x-axis label, y-axis
+#'   label, plot legend, watermark, respectively.
+#' @field legendPosition A character string defining the legend position
+#'   (default: `"outsideRight"`). Available options
+#'   are:`"none"`,`"insideTop"`,`"insideTopLeft"`,
+#'   `"insideLeft"`,`"insideBottomLeft"`,`"insideBottom"`,`"insideBottomRight"`,
+#'   `"insideRight"`,`"insideTopRight"`,`"outsideTop"`,`"outsideTopLeft"`,
+#'   `"outsideLeft"`,`"outsideBottomLeft"`,`"outsideBottom"`,`"outsideBottomRight"`,
+#'    `"outsideRight"`,`"outsideTopRight"`.
+#' @field legendTitleSize,legendTitleColor,legendTitleFontFamily,legendTitleFontFace,legendTitleAngle,legendCaptionSize Aesthetic properties for the legend title.
+#' @field legendCaptionColor,legendCaptionFontFamily,legendCaptionFontFace,legendCaptionAngle Aesthetic properties for the legend caption.
+#' @field xAxisTickLabels,xAxisLabelSize,xAxisLabelColor,xAxisLabelFontFamily,xAxisLabelFontFace,xAxisLabelAngle Aesthetic properties for the x-axis label.
+#' @field yAxisTickLabels,yAxisLabelSize,yAxisLabelColor,yAxisLabelFontFamily,yAxisLabelFontFace,yAxisLabelAngle Aesthetic properties for the y-axis label.
+#' @field xAxisLimits,yAxisLimits A numeric vector of axis limits for the x-and
+#'   y-axis, respectively.
+#' @field xAxisTicks,yAxisTicks A numeric vector or a function defining where to
+#'   position x-and y-axis ticks, respectively.
+#' @field xAxisScale,yAxisScale A character string defining axis scale (default:
+#'   `"lin"`). Available options are: `"lin"`, `"log"`, `"ln"`, `"discrete"`,
+#'   `"reverse"`, `"sqrt"`, `"time"`, `"date"`.
```

Comment:
add hint that all options are accessible via `tlf::Scaling` e.g. `tlf::Scaling$lin`
https://www.open-systems-pharmacology.org/TLF-Library/reference/LegendPositions.html
https://www.open-systems-pharmacology.org/TLF-Library/reference/Scaling.html